index.js

```diff
@@ -22,6 +22,8 @@ module.exports = declare((api, options) => {
 
   const debug = (options && typeof options.debug === 'boolean') ? !!options.debug : false;
 
+  const { devEnv = 'development' } = options || {};
```

Comment:
Defaulting to a global doesn't make sense. `api.env(devEnv)` is already reading the NODE_ENV and BABEL_ENV and Babel's `envName`. This is just choosing what to compare it to.

Reply:
ah, that wasn't clear. so what's the precedence here? If i set NODE_ENV to "production", does that win over a envName of "development"?

Reply:
There are several ways of setting the `envName` for `babel` -- ones I'm aware of are `BABEL_ENV`, `NODE_ENV`, and the `--envName` flag.

Reply:
yes but which beats which? having several ways means it's very unclear which one "wins" over which others

Reply:
I believe setting the `envName` flag just overrides `BABEL_ENV`, [which takes precedence over `NODE_ENV`](/babel/babel/blob/master/packages/babel-core/src/config/helpers/environment.js#L4).

Reply:
One passed to Babel overrides all, and BABEL_ENV overrides NODE_ENV, and when none are given, the default is `"development"`. Docs here: https://babeljs.io/docs/en/options#envname

Reply:
If you want to just do NODE_ENV you can always do 

```
api.cache.using(() => devEnv === process.env.NODE_ENV || "development")
```
Or something like it.

Reply:
ok so, I'd like `NODE_ENV` to always win over everything when `devEnv` is not provided to this preset, so i'd like this line to do `const { devEnv = process.env.NODE_ENV || 'development' } = options || {};`, as in the original comment

Reply:
I feel like we should take a step back here and get a full example? What are you actually expecting `devEnv` to be used as, if it we're initialized like that?

Reply:
I'd like users of babel-preset-airbnb to get a) the right behavior by default, which means "using NODE_ENV", and b) be able to override react's env setting independently.

Reply:
@ljharb I mean, do you want to be able to override a boolean of "is this development" or do you want to be able to override the name of the environment that qualifies as `development`?

Reply:
Perhaps it's better to just add a boolean option `development`, which will be passed down to the `development` option of `@babel/preset-react`. There doesn't seem to be any other use of the environment in this preset, so having a special case of it might be confusing.

People can then either use the `env` option (which uses `envName`) of the `json` `.babelrc` (which means they'd need to specify `babel-preset-airbnb` more than once to have that option be different) or use `process.env.NODE_ENV === 'development'` in `.babelrc.js` or `babel.config.js`.